### ISSUE TITLE
Support for common North American non-standard timezones

### ISSUE THREAD

**Issue by turingtest37** [JuliaTime/TimeZones.jl#299]
The legacy "non-standard" (i.e. out of fashion) timezones used in North America include "EDT" for (Eastern Daylight Time), "EST", "CST", etc. These are currently not supported when parsing datestrings.
```julia

julia> datestr = "Sun, 04 Oct 2020 12:02:25 EDT"
"Sun, 04 Oct 2020 12:02:25 EDT"
julia> zdt = parse(ZonedDateTime, datestr, dateformat"eee, dd u yyyy HH:MM:SS zzz")
ERROR: ArgumentError: Unable to parse date time. Expected directive DatePart(zzz) at char 27

```

I have worked around this for the EDT and EST timezones with a simple function that could be extended:

`normtz(datestr) =  replace(datestr, r"(.*)E([DS])T$" => (s"\1" == "D" ? s"\1-0400" : s"\1-0500"))`

See also the slack discussion:
https://julialang.slack.com/archives/C6A044SQH/p1601828472301400

**Comment by omus** [comment]
The main issue is that time zone abbreviations are used in many different time zones and in some cases refer to very different offsets. Take "CST" for example is used in "America/Chicago" to mean UTC-6 while in "Asia/Shanghai" it means UTC+8:

```julia
julia> using TimeZones

julia> ZonedDateTime(2020,3,1,tz"Asia/Shanghai").zone
CST (UTC+8)

julia> ZonedDateTime(2020,3,1,tz"America/Chicago").zone
CST (UTC-6)
```
One possible solution to this problem would be to have the caller pass in the set of time zones they expect to parse:
```julia
julia> parse(ZonedDateTime, "Sun, 04 Oct 2020 12:02:25 EDT", df, timezones=[FixedTimeZone("EDT", -6 * 3600, 1 * 3600)])
```
I could make the set of North American time zones as a constant in TimeZones.jl to make easier.

**Comment by turingtest37** [comment]
Yes! That is an elegant way of doing the regex-based replace that worked for me. And it should (if it doesn't already) support knowing at what time these traditional time zones change for a given zone, e.g. when EST reverts to EDT in October of each year.
